aptos-move/framework/aptos-framework/sources/account/common_account_abstractions/derivable_account_abstraction_solana.move

```diff
@@ -0,0 +1,249 @@
+/// Domain account abstraction using ed25519 hex for signing.
+///
+/// Authentication takes digest, converts to hex (prefixed with 0x, with lowercase letters),
+/// and then expects that to be signed.
+/// authenticator is expected to be signature: vector<u8>
+/// account_identity is raw public_key.
+module aptos_framework::derivable_account_abstraction_ed25519_hex {
+    use aptos_framework::auth_data::AbstractionAuthData;
+    use aptos_std::ed25519::{
+        Self,
+        new_signature_from_bytes,
+        new_unvalidated_public_key_from_bytes,
+    };
+    use std::chain_id;
+    use std::error;
+    use std::string_utils;
+    use std::transaction_context::{Self, EntryFunctionPayload};
+
+    const EINVALID_SIGNATURE: u64 = 1;
+    const EINVALID_BASE_58_PUBLIC_KEY: u64 = 2;
+    const EUNSUPPORTED_CHAIN_ID: u64 = 3;
+    const EMISSING_ENTRY_FUNCTION_PAYLOAD: u64 = 4;
+
+    const BASE_58_ALPHABET: vector<u8> = b"123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+    const HEX_ALPHABET: vector<u8> = b"0123456789abcdef";
+
+    fun split_abstract_public_key(abstract_public_key: &vector<u8>): (vector<u8>, vector<u8>) {
+        // First 44 bytes are the base58 utf8 encoded public key
+        let base58_public_key = abstract_public_key.slice(0, 44);
+        let domain = abstract_public_key.slice(44, abstract_public_key.length());
+        (base58_public_key, domain)
+    }
+
+    fun network_name(): vector<u8> {
+        let chain_id = chain_id::get();
+        if (chain_id == 1) {
+            b"mainnet"
+        } else if (chain_id == 2) {
+            b"testnet"
+        } else if chain_id == 3 {
+            b"devnet"
+        } else if chain_id == 4 {
+            b"localnet"
```

Comment:
sorry, in the ts-sdk it is called "local" and thats what we use in the adapter
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/utils/apiEndpoints.ts#L79
```suggestion
            b"local"
```